src/smolagents/tools.py

```diff
@@ -413,30 +413,25 @@ def from_hub(
         )
 
         tool_code = Path(tool_file).read_text()
+        return Tool.from_code(tool_code, **kwargs)
+
+    @classmethod
+    def from_code(cls, tool_code: str, **kwargs):
+        module = types.ModuleType("dynamic_tool")
+
+        print(tool_code)
+
+        exec(tool_code, module.__dict__)
+
+        # Find the Tool subclass
+        tool_class = None
+        for item in module.__dict__.values():
+            if isinstance(item, type) and issubclass(item, Tool) and item != Tool:
+                tool_class = item
+                break
```

Comment:
```suggestion
        tool_class = next(
            (
                obj
                for _, obj in inspect.getmembers(module, inspect.isclass)
                if issubclass(obj, Tool) and obj is not Tool
            ),
            None,
        )
```
`__dict__.values()` loops over everything, while getmembers only selects classes. Safer/faster.